src/main/java/com/github/wojtechm/zadania_rekrutacyjne/string/Reduce.java

```diff
@@ -19,7 +19,18 @@
 @Difficulty(Level.MEDIUM)
 class Reduce {
 
+    private Reduce(){}
+
+    public static void main(String[] args) {
+        System.out.println(reduce("book"));
+    }
     static String reduce(String toReduce) {
+        if(toReduce == null) throw new IllegalArgumentException("Cannot reduce null");
+        var lastLength = -1;
+        while (lastLength != toReduce.length()) {
+            lastLength = toReduce.length();
+            toReduce = toReduce.replaceAll("(.)\\1", "");
```

Comment:
Cool idea, but it's O(n**2). Could you please implement it so it has linear complexity?

Reply:
Maybe done